modules/extensions-support/src/main/java/org/mule/runtime/module/extension/internal/metadata/MetadataMediator.java

```diff
@@ -189,21 +203,30 @@ public MetadataMediator(T componentModel) {
 
     MetadataResult<OutputMetadataDescriptor> output = outputDelegate.getOutputMetadataDescriptor(context, keyValue);
     MetadataResult<InputMetadataDescriptor> input = inputDelegate.getInputMetadataDescriptors(context, keyValue);
-
-    if (output.isSuccess() && input.isSuccess()) {
+    Optional<MetadataResult<InputMetadataDescriptor>> successCallbackInput = successCallbackInputDelegate
+        .map(successCallbackInputDelegate -> successCallbackInputDelegate.getInputMetadataDescriptors(context, keyValue));
+    Optional<MetadataResult<InputMetadataDescriptor>> errorCallbackInput = errorCallbackInputDelegate
+        .map(errorCallbackInputDelegate -> errorCallbackInputDelegate.getInputMetadataDescriptors(context, keyValue));
```

Comment:
![Code Smell](https://sonarsource.github.io/sonar-github/code_smell.png 'Code Smell') Code Smell: Rename "errorCallbackInputDelegate" which hides the field declared at line 86. ([squid:HiddenFieldCheck](https://sonarqube.kbuild.msap.io/coding_rules#rule_key=squid%3AHiddenFieldCheck))

[See it in SonarQube](https://sonarqube.kbuild.msap.io/project/issues?id=mule-runtime-mule&pullRequest=7668&issues=AWjdfEMLotIXnmsUGfbV&open=AWjdfEMLotIXnmsUGfbV)

[ISSUE_KEY]: AWjdfEMLotIXnmsUGfbV